src/ol/controls/RoutingControl.js

```diff
@@ -0,0 +1,475 @@
+      const turfLineSegment = turfLineSlice(
+        turfPoint(viaPointClosestPoints[i]),
+        turfPoint(viaPointClosestPoints[i + 1]),
+        turfLineString(oldRoute.getGeometry().getCoordinates()),
+      );
+
+      /* Create an ol LineString from segment and get the distance between
+       * the click point and the closest point on segment
+       */
+      const segment = new LineString(turfLineSegment.geometry.coordinates);
+      const distanceToClick = getDistance(
+        segment.getClosestPoint(this.initialRouteDrag.coordinate),
+        this.initialRouteDrag.coordinate,
+      );
+
+      // Return segment index with smallest distance to click point
+      if (!distanceToSegment || distanceToClick < distanceToSegment) {
+        distanceToSegment = distanceToClick;
+        newViaIndex = i + 1;
+      }
+    }
+
+    // Insert new viaPoint at given index
+    return this.addViaPoint(e.mapBrowserEvent.coordinate, newViaIndex);
+  }
+
+  createDefaultElement() {
+    /**
+     * Define a default element.
+     */
+    this.element = document.createElement('button');
+    this.element.id = 'ol-toggle-routing';
+    this.element.innerHTML = 'Toggle Route Control';
+    this.element.onclick = () =>
+      this.active ? this.deactivate() : this.activate();
+    Object.assign(this.element.style, {
+      position: 'absolute',
+      right: '10px',
+      top: '10px',
+    });
+  }
+
+  /**
+   * Create the interaction used to modify vertexes of features.
+   * @private
+   */
+  createModifyInteraction() {
+    /**
+     * @type {ol.interaction.Modify}
+     * @private
+     */
+    // Define and add modify interaction
+    this.modifyInteraction = new Modify({
+      source: this.routingLayer.olLayer.getSource(),
+      pixelTolerance: 4,
+      hitDetection: this.routingLayer.olLayer,
+      deleteCondition: (e) => {
+        const feats = e.target.getFeaturesAtPixel(e.pixel, {
+          hitTolerance: 5,
+        });
+        const viaPoint = feats.find(
+          (feat) => feat.getGeometry() instanceof Point && feat.get('index'),
+        );
+        if (click(e) && viaPoint) {
+          // Remove node & viaPoint if an existing viaPoint was clicked
+          this.removeViaPoint(viaPoint.get('index'));
+          return true;
+        }
+        return false;
+      },
+    });
+    this.modifyInteraction.on('modifystart', this.onModifyStart);
+    this.modifyInteraction.on('modifyend', this.onModifyEnd);
+    this.modifyInteraction.setActive(false);
+  }
+
+  /**
+   * Add click listener to map.
+   * @private
+   */
+  addListeners() {
+    this.removeListeners();
+    this.onMapClickKey = this.map.on('singleclick', this.onMapClick);
+  }
+
+  /**
+   * Remove click listener from map.
+   * @private
+   */
+  removeListeners() {
+    unByKey(this.onMapClickKey);
+  }
+
+  activate() {
+    super.activate();
+    if (
+      this.map &&
+      !this.map.getInteractions().getArray().includes(this.modifyInteraction)
+    ) {
```

Comment:
you shouldn't have to do this 'if' you can remove it .

Reply:
without the condition it will add the same interaction again I think no?